### DISCUSSION TITLE
Different build directories for different OSes?

### DISCUSSION THREAD

**Discussion by aubade** [microsoft/vscode-cmake-tools#3580] [Q&A] [answered]
I have the same source tree that I want to build my project on in different OSes. I'd like to do some sort of variable substitution in the cmake.buildDirectory setting--something along the lines of ${workspaceFolder}/build-${OPERATING_SYSTEM}-${variant:buildType}

So it'd result in e.g. project/build-Linux-release or project/build-Windows-debug

But I can't find anything like this after a couple hours of searching. Is there a way to do this, or am I out of luck?

**Comment by csorvagep** [comment, marked as answer]
Hi,

Not sure if it applies for your use-case, but you could try [CmakePresets](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md) for this. Its easy to setup, but if you frequently switch between the different presets, then might not be the best for you.
